tests/x11/apache_ssl.pm

```diff
@@ -0,0 +1,76 @@
+# SUSE's Apache+SSL tests
+#
+# Copyright © 2016 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Enable SSL module on Apache2 server
+# Maintainer: Qingming Su <qingming.su@suse.com>
+
+use base "x11test";
+use testapi;
+use utils;
+use strict;
+
+sub quit_firefox() {
+    send_key "alt-f4";
+    if (check_screen("firefox-save-and-quit", 10)) {
+        send_key "ret";
+    }
+}
+
+sub setup_apache_ssl() {
+    select_console 'root-console';
+
+    # Make sure the apache2 package is installed
+    assert_script_run 'zypper -n in apache2', 180;
+
+    # Enable the ssl module for apache
+    assert_script_run 'a2enmod ssl';
+    assert_script_run "sed -i '/^APACHE_SERVER_FLAGS=/s/^/#/' /etc/sysconfig/apache2";
+    assert_script_run 'echo APACHE_SERVER_FLAGS="SSL" >> /etc/sysconfig/apache2';
+
+    # Create x509 certificate for this apache server
+    assert_script_run 'gensslcert -n localhost -C localhost -e webmaster@localhost', 300;
+    assert_script_run 'ls /etc/apache2/ssl.crt/localhost-server.crt /etc/apache2/ssl.key/localhost-server.key';
+
+    # Prepare vhost file
+    assert_script_run 'cp /etc/apache2/vhosts.d/vhost-ssl.template /etc/apache2/vhosts.d/vhost-ssl.conf';
+    assert_script_run "sed -i -e 's/vhost-example.crt/localhost-server.crt/g' -e 's/vhost-example.key/localhost-server.key/g' /etc/apache2/vhosts.d/vhost-ssl.conf";
+
+    # Start apache service
+    assert_script_run 'systemctl start apache2';
+    assert_script_run 'systemctl is-active apache2';
+
+    # Create a test html page
+    assert_script_run 'echo "<html><h2>Hello Linux</h2></html>" > /srv/www/htdocs/hello.html';
+}
+
+sub run() {
+    my $self = shift;
+    $self->setup_apache_ssl();
+
+    # Start firefox and load the testing page
+    select_console 'x11';
+    ensure_unlocked_desktop;
+    x11_start_program("firefox https://localhost/hello.html");
+    assert_screen "firefox-ssl-untrusted", 60;
+    assert_and_click "firefox-ssl-advanced";
+    assert_and_click "firefox-ssl-addexception";
+    assert_screen "firefox-ssl-confirm-exception", 60;
+    send_key "alt-c";
+    assert_screen 'firefox-ssl-load-hellopage', 60;
```

Comment:
Yeah, using curl command is more lightweight and easier to implement.
But I thought using firefox was more visible for end user to check the https process with untrusted certificate.